kv/dist_sender.go

```diff
@@ -544,6 +597,26 @@ func (ds *DistSender) Send(call *client.Call) {
 	}
 }
 
+// updateLeaderCache updates the cached leader for the given Raft group,
+// evicting any previous value in the process.
+// The new leader is cached only if it isn't equal to the newly evicted value.
+func (ds *DistSender) updateLeaderCache(rid proto.RaftID, leader proto.Replica) {
+	// Slight overhead here since we want to make sure that when the new
+	// proposed leader equals the old, we only evict.
+	oldLeader := ds.leaderCache.Lookup(rid)
+	ds.leaderCache.Update(rid, nil)
+	if oldLeader != nil {
+		log.V(1).Infof("raft %d: evicted cached leader %d", rid, oldLeader.StoreID)
+	}
+	if leader.StoreID == 0 {
+		return
+	}
+	if oldLeader == nil || leader.StoreID != oldLeader.StoreID {
+		log.V(1).Infof("raft %d: new cached leader %d", rid, leader.StoreID)
+		ds.leaderCache.Update(rid, &leader)
```

Comment:
Why store a pointer to the leader? Just store a copy of the replica instead.

Also, I don't see where we're going to be evicting here. What am I missing?